src/sysrepo.h

```diff
@@ -247,13 +252,22 @@ int sr_connect(const sr_conn_options_t opts, sr_conn_ctx_t **conn);
 int sr_disconnect(sr_conn_ctx_t *conn);
 
 /**
- * @brief Learn the current global number of (some possibly dead) connections.
+ * @brief Learn the current global number of alive connections.
  *
  * @param[out] conn_count Current number of connections.
  * @return Error code (::SR_ERR_OK on success).
  */
 int sr_connection_count(uint32_t *conn_count);
 
+/**
+ * @brief Retrieve globally unique connection ID for this connection.
+ * A value of 0 indicates the ID is not set.
+ *
+ * @param[in] conn Connection to use.
+ * @return unique identifier for the connection.
+ */
+sr_cid_t sr_connection_cid(sr_conn_ctx_t *conn);
```

Comment:
Not sure why should this function be public, using it in the tests is not a reason good enough.

Reply:
You are correct, there really is not a good reason to expose this beyond its use in the tests.  That being said, I did not want to eliminate the tests which verified the pid and did not have a good way to update/modify the monitoring xml.  Any suggestions?

Reply:
What I did in some other test (not sure if this project, it does not matter) is to manually zero values I want to test before comparison. However, there was a fixed length unlike this use-case so it may not really be possible. Another solution I can think of right now is to use a special function for comparing this that will skip actual `cid` value and it will be missing in the expected XML.

Reply:
I implemented something similar by deleting content in the <cid>...</cid> and <notification-sub>...</notification-sub> tags. This allows comparison of the response to a fixed string.